client/src/LayoutComponents/ProjectDetails/ProjectDetails.js

```diff
@@ -0,0 +1,101 @@
+import React, { Component } from 'react';
+import './ProjectDetails.css';
+import github from '../../images/github.svg';
+import pc from '../../images/pc.svg';
+import ProjectView from '../../SharedComponents/ProjectView';
+import StudentCard from '../../SharedComponents/StudentCard';
+import { Link } from 'react-router-dom';
+import axios from 'axios';
+class ProjectDetails extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      projects: [],
+    };
+  }
+  componentDidMount() {
+    const { id } = this.props;
+    // id coming from cohorts and studentProfile pages
+    axios
+      .get(`api/project/${id}`)
+      .then((res) => {
+        const resultArray = res.data.map((e) => e);
+        this.setState({
+          projects: resultArray,
+        });
+      })
+      .catch((error) => error.response);
+  }
+  render() {
+    const { projects } = this.state;
+
+    return (
+      <div className='projectdetails-container'>
+        {!projects.length ? (
+          <div className='loading-container'>
```

Comment:
@kholoudfann If you want to reuse this same loading spinner in other parts of the app, you could refactor it out into its own component and then import and use it in this component.

Reply:
thanks ,  i will do it on new branch ,  Is the page now good and code good?

Reply:
Sounds good! Yes, it all looks good to me.